### PR TITLE
Remove `CycleEngine`

### PR DESCRIPTION
**Pull request by weikengchen**
## Description

In the last release, CycleEngine has been marked as deprecated. For the incoming release, `CycleEngine` will be removed.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

N/A:
- [ ] Wrote unit tests

### CODE REVIEW

**Comment by weikengchen** [comment]
Updated the CHANGELOG. Also, remove a duplicated entry about #279.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +15,6 @@
 - [\#261](https://github.com/arkworks-rs/algebra/pull/261) (ark-ff) Add support for 448-bit integers and fields.
 - [\#263](https://github.com/arkworks-rs/algebra/pull/263) (ark-ff) Add `From<iXXX>` implementations to fields.
 - [\#265](https://github.com/arkworks-rs/algebra/pull/265) (ark-serialize) Add hashing as an extension trait of `CanonicalSerialize`.
-- [\#279](https://github.com/arkworks-rs/algebra/pull/279) (ark-ec) Parallelize miller loop operations for BLS12.
```
Comment (Pratyush) [inline CHANGELOG.md:16]: Why is this deleted?

File: ec/README.md [hunk 2]
```diff
@@ -14,7 +14,6 @@ The available elliptic curve traits are:
 * [`AffineCurve`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L223) - Interface for elliptic curve points in the 'canonical form' for serialization.
 * [`ProjectiveCurve`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L121) - Interface for elliptic curve points in a representation that is more efficient for most computation.
 * [`PairingEngine`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L44) - Pairing friendly elliptic curves (Contains the pairing function, and acts as a wrapper type on G1, G2, GT, and the relevant fields).
-* [`CycleEngine`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L318) - Wrapper type for a cycle of pairing friendly elliptic curves.
```
Comment (Pratyush) [inline ec/README.md:17]: Didn't we replace this with `CurveCycle` and `PairingFriendlyCycle`? Maybe instead of deleting it we should replace `CycleEngine` with this?

File: CHANGELOG.md (hunk 1 above)
Comment (weikengchen) [reply CHANGELOG.md:16]: #279 (ark-ec) Parallelize miller loop operations for BLS12. is also listed as an "improvement" above.

File: ec/README.md (hunk 2 above)
Comment (weikengchen) [reply ec/README.md:17]: Good points. Let me update the README

Comment (weikengchen) [reply ec/README.md:17]: Any feedback? I added two lines and just also updated the line numbers.
```
* [`CurveCycle`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L319) - Wrapper type for a cycle of elliptic curves.
* [`PairingFriendlyCycle`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L331) - Wrapper type for a cycle of pairing friendly elliptic curves. 
```

Comment (Pratyush) [reply ec/README.md:17]: Maybe say "Trait representing a cycle of..."?

Comment (weikengchen) [reply ec/README.md:17]: Done!

**Review by Pratyush — APPROVED** [review]
